hub/client/client.py

```diff
@@ -0,0 +1,122 @@
+import requests
+import sys
+from hub.client.config import (
+    HUB_REST_ENDPOINT,
+    GET_TOKEN_SUFFIX,
+    REGISTER_USER_SUFFIX,
+    DEFAULT_REQUEST_TIMEOUT,
+)
+from hub.client.utils import get_auth_header, check_response_status
+from hub.util.exceptions import LoginException
+from typing import Optional
+from hub import __version__
+
+
+class HubBackendClient:
```

Comment:
What is the scope of this class, and why does it exist? What state does it maintain?

We might be better off with a couple independent low-level functions, and then we can make a meaningful wrapper on top if we need a higher-level interface.

Reply:
So this class essentially will get reused for the dataset object. Whenever a dataset needs to interact with backend i.e. for fetching creds, updating dataset state, etc it'll use this. It only maintains the auth_header which is read from a file in the local fs. This class prevents that read operation from happening multiple times I guess but that's not a big overhead though. We can split it into functions as well but having a single client object to use for sending all kinds of requests personally feels neater to me.